lib/ecto/multi.ex

```diff
@@ -274,17 +274,17 @@ defmodule Ecto.Multi do
   The function should return either `{:ok, value}` or `{:error, value}`, and
   receives changes so far as an argument.
   """
-  @spec run(t, name, run) :: t
-  def run(multi, name, run) when is_function(run, 1) do
-    add_operation(multi, name, {:run, run})
+  @spec run(t, name, fun) :: t
```

Comment:
The current value fo the `run` type expresses the value that's stored in the `Ecto.Multi` struct. The value accepted by the function is different, so we should use different types.

Since `run` is a public type, we should change it to match the value accepted by the function - i.e. `(t -> {:ok | :error, any})`. We should additionally update the `operation` type to include the correct value for run -> `{:run, run | {module, atom, [any]}}`